### PR TITLE
Add missing attributes to `Repository` 

### PR DESCRIPTION
**Pull request by RobPasMue**
According to https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#update-a-repository

New parameters are accepted and others have been removed.

Parameters included in this PR:
``visibility``
``is_template``
``use_squash_pr_title_as_default``
``squash_merge_commit_title``
``squash_merge_commit_message``
``merge_commit_title``
``merge_commit_message``
``web_commit_signoff_required``

Paramters removed in this PR:
``has_downloads``

Parameters have also been ordered according to the REST API order in the documentation

### CODE REVIEW

**Comment by RobPasMue** [comment]
Hi @trim21 ! Let me know if I need to do anything else to get this feature merged :) I hope I did things properly

File: github/Repository.py [hunk 1]
```diff
@@ -1772,6 +1774,7 @@ def edit(
         assert description is github.GithubObject.NotSet or isinstance(description, str), description
         assert homepage is github.GithubObject.NotSet or isinstance(homepage, str), homepage
         assert private is github.GithubObject.NotSet or isinstance(private, bool), private
+        assert visibility is github.GithubObject.NotSet or isinstance(visibility, str), visibility
```
Comment (EnricoMi) [inline github/Repository.py:1777]: Can we assert here `visibility` has only the allowed values?
```suggestion
        assert visibility is github.GithubObject.NotSet or visibility in [
            "public", "private"
        ], visibility
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
While you are adding a missing attribute, can you add those that are also missing, e.g.
- is_template
- use_squash_pr_title_as_default
- squash_merge_commit_title
- squash_merge_commit_message
- merge_commit_title
- merge_commit_message
- web_commit_signoff_required

Please check if I missed some.

**Comment by RobPasMue** [comment]
Yep, for sure @EnricoMi - let me do that! =)

File: tests/Repository.py [hunk 2]
```diff
@@ -134,7 +134,6 @@ def testEditWithAllArguments(self):
             has_issues=True,
```
Comment (EnricoMi) [inline tests/Repository.py:134]: Can we test will all new arguments? Recording test data should be easy to do.

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Can we get add all those new attributes to `_initAttributes` and `_useAttributes` and create getters as well? Please record test data for `testAttributes` and add those attributes there as well.

**Comment by RobPasMue** [comment]
Hi @EnricoMi - issues addressed. I added the properties, initialized the attributes and modified the useAttributes function. I also updated the test data. I hope everything is fine now. Let me know if you need anything else

**Comment by MaxJPRey** [comment]
Nice work @RobPasMue . Thanks for updating the properties according to the API.

**Review by EnricoMi — APPROVED** [review]
Awesome, good work!

**Pull request merged**